pkg/controllers/provisioning/scheduling/node.go

```diff
@@ -27,65 +29,103 @@ import (
 // Node is a set of constraints, compatible pods, and possible instance types that could fulfill these constraints. This
 // will be turned into one or more actual node instances within the cluster after bin packing.
 type Node struct {
-	Constraints         *v1alpha5.Constraints
+	v1alpha5.Constraints
 	InstanceTypeOptions []cloudprovider.InstanceType
 	Pods                []*v1.Pod
-	unschedulable       bool // if true, the node will accept no more pods
+
+	podResources    v1.ResourceList
+	daemonResources v1.ResourceList
 }
 
-func NewNode(constraints *v1alpha5.Constraints, instanceTypeOptions []cloudprovider.InstanceType, pods ...*v1.Pod) *Node {
-	n := &Node{Constraints: constraints.DeepCopy()}
+func NewNode(constraints *v1alpha5.Constraints, daemonResources v1.ResourceList, instanceTypeOptions []cloudprovider.InstanceType) *Node {
+	n := &Node{
+		Constraints:     *constraints.DeepCopy(),
+		daemonResources: daemonResources,
+	}
 
 	for _, it := range instanceTypeOptions {
+		// If a zero-resource pod can't fit, don't consider this instance type.  This occurs if the node overhead +
+		// daemon set requests is already larger than the instance type can support
+		if !n.newPodCanFit(nil, it) {
```

Comment:
This would only cause an issue if we tried to write to the nil map, reading from it is the same as reading from an empty map.  I can change it, but it's 4x slower to pass an empty map:


```go
package main

import "testing"

type ResourceName string
type Quantity struct {
	i int64
}
type ResourceList map[ResourceName]Quantity

func foo(m ResourceList) int64 {
	return m["test"].i
}
func BenchmarkNilMap(b *testing.B) {
	for i := 0; i < b.N; i++ {
		foo(nil)
	}
}
func BenchmarkEmptyMap(b *testing.B) {
	for i := 0; i < b.N; i++ {
		foo(ResourceList{})
	}
}
```
```
BenchmarkNilMap-12      	678223546	         1.774 ns/op
BenchmarkEmptyMap-12    	169209038	         7.071 ns/op
```